packages/expo-firebase-analytics/ios/EXFirebaseAnalytics/EXFirebaseAnalytics.m

```diff
@@ -0,0 +1,213 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+#import <UMCore/UMUtilities.h>
+#import <EXFirebaseAnalytics/EXFirebaseAnalytics.h>
+#import <UIKit/UIKit.h>
+#import <Firebase/Firebase.h>
+
+static NSString *const DEFAULT_APP_DISPLAY_NAME = @"[DEFAULT]";
+static NSString *const DEFAULT_APP_NAME = @"__FIRAPP_DEFAULT";
+
+@implementation EXFirebaseAnalytics
+
+UM_EXPORT_MODULE(ExpoFirebaseAnalytics);
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+- (void)rejectException:(UMPromiseRejectBlock)reject exception:(NSException *)exception {
+  NSError *error = [NSError errorWithDomain:@"ERR_FIREBASE_ANALYTICS" code:4815162342 userInfo:@{
+        @"message": exception.reason,
+        @"code": exception.name,
+    }];
+  reject(exception.name, exception.reason, error);
+}
+
+UM_EXPORT_METHOD_AS(initializeAppDangerously,
+                    initializeAppDangerously:(NSDictionary *)config
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject) {
+  [UMUtilities performSynchronouslyOnMainThread:^{
```

Comment:
If `method_queue` is overridden and returns main queue it would be a really bad bug if exported methods wouldn't be run on main thread